### PR TITLE
Apollo87H PCB support

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for the Apollo87H keyboard, revisions Gamma and Delta.

_Two other versions of Apollo, 87H-T-SC and 88H-T-SC are mentioned in the upfolder README and their firmwares will be PR'd once the prototypes are tested._ (EDIT: both new variants were added in in https://github.com/qmk/qmk_firmware/commit/27a7ac2451c1de3d5559b4efa730c872de5c7e56 and
https://github.com/qmk/qmk_firmware/commit/eca6b80a5f300cb2147992bb36dd7e27b7e83f37 ).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/acheron/apollo/readme.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Acheron Apollo QMK firmware
+
+![apollo](https://acheronproject.com/images/apollo/delta-pictures/delta1.jpeg)
```
Comment (fauxpark) [inline keyboards/acheron/apollo/readme.md:3]: Please use a smaller image here.

File: keyboards/acheron/apollo/87h/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Acheron Apollo87H QMK firmware
+
+![apollo](https://acheronproject.com/images/apollo/delta-pictures/delta1.jpeg)
```
Comment (fauxpark) [inline keyboards/acheron/apollo/87h/readme.md:3]: As above.

File: keyboards/acheron/apollo/87h/gamma/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+RGB_MATRIX_ENABLE = yes      # Enable keyboard RGB underglow
+RGB_MATRIX_DRIVER = IS31FL3741
+
+KEYBOARD_SHARED_EP = yes
+
+AUDIO_ENABLE = no          # Audio output
+EEPROM_DRIVER = i2c
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
+STM32_BOOTLOADER_ADDRESS = 0x1FFF0000
```
Comment (fauxpark) [inline keyboards/acheron/apollo/87h/gamma/rules.mk:23]:
```suggestion
# MCU name
MCU = STM32F411

# Bootloader selection
BOOTLOADER = stm32-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = IS31FL3741
KEYBOARD_SHARED_EP = yes
EEPROM_DRIVER = i2c

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
```

File: keyboards/acheron/apollo/87h/delta/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = no           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGB_MATRIX_ENABLE = yes      # Enable keyboard RGB underglow
+RGB_MATRIX_DRIVER = WS2812
+
+AUDIO_ENABLE = no          # Audio output
+EEPROM_DRIVER = i2c
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
+STM32_BOOTLOADER_ADDRESS = 0x1FFF0000
```
Comment (fauxpark) [inline keyboards/acheron/apollo/87h/delta/rules.mk:21]:
```suggestion
# MCU name
MCU = STM32F401

# Bootloader selection
BOOTLOADER = stm32-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
EEPROM_DRIVER = i2c

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE -DDEBUG_EEPROM_OUTPUT=TRUE
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Gondolindrim** [comment]
Took the liberty of adding the missing 87H-T-SC and 88H-T-SC variants since the PR was not closed yet

**Comment by Gondolindrim** [comment]
Took the opportunity to add the missing 87H-T-SC and 88H-T-SC variants in  27a7ac2 and 
eca6b80

File: keyboards/acheron/apollo/87h/delta/chconf.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
```
Comment (drashna) [inline keyboards/acheron/apollo/87h/delta/chconf.h:17]: This this file needed?

File: keyboards/acheron/apollo/87h/gamma/chconf.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
```
Comment (drashna) [inline keyboards/acheron/apollo/87h/gamma/chconf.h:17]: Is this file needed?

Comment (Gondolindrim) [reply keyboards/acheron/apollo/87h/gamma/chconf.h]: Removed in [9a67dbd](https://github.com/qmk/qmk_firmware/pull/16817/commits/9a67dbddc06a0e5cd3c7c38ff925ae18c1392e50)

File: keyboards/acheron/apollo/87h/delta/chconf.h (hunk 5 above)
Comment (Gondolindrim) [reply keyboards/acheron/apollo/87h/delta/chconf.h]: Removed in [9a67dbd](https://github.com/qmk/qmk_firmware/pull/16817/commits/9a67dbddc06a0e5cd3c7c38ff925ae18c1392e50)

**Review by noroadsleft — CHANGES_REQUESTED** [review]
All of these boards implement layouts that are part of the Community Layouts feature. It would be much preferred if you could use the layout macro names that are part of that feature.

Applying this `.diff` file will rename the layout macros as desired (saves me posting about 30-something review comments).

[apollo_87_series_layout_names.diff.txt](https://github.com/qmk/qmk_firmware/files/9107053/apollo_87_series_layout_names.diff.txt)

**Comment by Gondolindrim** [comment]
> All of these boards implement layouts that are part of the Community Layouts feature. It would be much preferred if you could use the layout macro names that are part of that feature.
> 
> Applying this `.diff` file will rename the layout macros as desired (saves me posting about 30-something review comments).
> 
> [apollo_87_series_layout_names.diff.txt](https://github.com/qmk/qmk_firmware/files/9107053/apollo_87_series_layout_names.diff.txt)

Sorry for the delay, had to test all three PCBs and tests check. Diff file was applied and committed in f047e08

**Review by noroadsleft — COMMENTED** [review]
@Gondolindrim,

Is the lack of `readme.md` files for the 87H-T-SC and 88H-T-SC deliberate? That and the lack of any `info.json` files are the only things I'm not satisfied with here, but I'm fine with authoring the `info.json` files for a follow-up PR.

**Comment by Gondolindrim** [comment]
> @Gondolindrim,
> 
> Is the lack of `readme.md` files for the 87H-T-SC and 88H-T-SC deliberate? That and the lack of any `info.json` files are the only things I'm not satisfied with here, but I'm fine with authoring the `info.json` files for a follow-up PR.

No particular reason other than I have no good pictures of the other two variants. The root `readme.md` already has the flashing details and layout information; there's also the documentation page. It just sounded redundant to have individual README's since they are all the same PCBs, even hardware-wise, with minor changes to reflect the different layouts.

If you think it would be a good deal to have README's for each variant I can add some.

As for the `info.json` for the configurator I just find myself too dumb to make it, I never get it right

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!
